docs/defaulting.md

```diff
@@ -0,0 +1,180 @@
+    type: qemu
+```
+
+This can be translated to a quite small Domain XML:
+
+```xml
+<domain type="qemu" xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
+  <name>testvm</name>
+  <memory unit="MB">64</memory>
+  <os>
+    <type>hvm</type>
+  </os>
+  <devices>
+      <interface type="network">
+      <source network="default"></source>
+    </interface>
+   </devices>
+</domain>
+```
+
+After we have defined the domain in libvirt, the XML looks like this:
+
+```xml
+<domain type='qemu' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
+  <name>testvm</name>
+  <uuid>a8335812-53d4-4e57-ac0b-72951c885a57</uuid>
+  <memory unit='KiB'>62500</memory>
+  <currentMemory unit='KiB'>62500</currentMemory>
+  <vcpu placement='static'>1</vcpu>
+  <os>
+    <type arch='x86_64' machine='pc-i440fx-2.7'>hvm</type>
+    <boot dev='hd'/>
+  </os>
+  <clock offset='utc'/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>destroy</on_crash>
+  <devices>
+    <emulator>/usr/bin/qemu-system-x86_64</emulator>
+    <controller type='usb' index='0' model='piix3-uhci'>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x2'/>
+    </controller>
+    <controller type='pci' index='0' model='pci-root'/>
+    <interface type='network'>
+      <mac address='52:54:00:2e:84:30'/>
+      <source network='default'/>
+      <model type='rtl8139'/>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x0'/>
+    </interface>
+    <input type='mouse' bus='ps2'/>
+    <input type='keyboard' bus='ps2'/>
+    <memballoon model='virtio'>
+      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
+    </memballoon>
+  </devices>
+</domain>
+```
+
+Finally, when a Domain was successfully started by Libvirt, Libvirt assigns an
+`alias` to every device in the XML.
+
+## The Problem
+
+ * Hotplug operations are not possible with the minimal VM spec of kubevirt,
+   since we could match wrong devices.
+ * After VM restarts, different devices or addresses could be assigned to a VM
+   by libvirt
+ * Libvirt assigns all that data on the host, either when defining the Domain,
+   or after the Domain was started. That requires asynchronous lookups, and
+   when thinking about administrating offline VMs, it binds your domain
+   specification management to the kubevirt runtime.
+
+If your VMs are ephemeral (no migrations, recreating VMs vs. restarting VMs, no
+hotplug), you are not affected by that.
+
+## Proposed Solution
+
+### Defaulting service based on libvirt
+
+Except for some runtime information, which is normally not included in
+migratable domain XMLs, libvirt fills in all defaults when defining a domain.
+We can leverage that with a simple REST based service, which defines VMs in an
+isolated libvirt (running libvirt and the small REST service in a Pod). The
+service is a standalone component in KubeVirt and can then be integrated into
+two different flows:
+
```

Comment:
then why is it needed? I still find the concept of a "libvirt dry run" to fill a host-specific default weird. Especially since not all the things are filled by libvirt (e.g. the example with qxl and spice - that's not done by libvirt)